### PR TITLE
promtail: add "max_age" field to configure cutoff for journal reading

### LINKED ISSUE #918: journal reading: Timestamp too old errors
Because the journal reader starts from the very beginning of the journal (i.e., not even the current boot, but the very first entry in the journal), promtail may end up sending extremely old logs to ingesters and getting errors back:

```level=error ts=2019-08-19T18:59:05.693794932Z caller=client.go:185 component=client host=localhost:3100 msg="final error sending batch" status=400 error="server returned HTTP status 400 Bad Request (400): sample for 'logs' has timestamp too old: 1562712027507"```

These errors will go away once the positions file is updated, but will still appear when a user uses journal reading in promtail for the first time. 

To work around this, the journal config should be updated to support a new field called `from` that takes a relative timestamp: 

```scrape_configs:
- job_name: journal
  journal:
    # read entries no older than 6h 
    from: 6h 
    path: /var/log/journal
    labels:
      job: systemd-journal
  relabel_configs:
    - source_labels: ['__journal__systemd_unit']
      target_label: 'unit'
```

## Implementation 

With the `from` field present, the journal reader has to do the following: 

1. If a positions file is present:
  1. Get journal entry at cursor.
  2. If entry's date is older than the allowed time specified by `from`, use the `from` date to initialize the journal entry. Entries will be lost. 
  3. If entry's date is newer than the allowed time specified by `from`, use the cursor as the start point. No entries will be lost. 
2. If no position file is present, just use the `from` date to initialize the journal reader. Old entries will not be imported into Loki.

### PR DESCRIPTION
**Pull request by rfratto**
The journal scrape config in promtail has been updated to support a "`max_age`" field. `max_age` determines the oldest journal entry promtail will read when starting the journal reader. When unspecified, `max_age` defaults to 7h.

Even if a position in the journal is saved in the promtail positions file, that position may be ignored if the entry corresponding to that position is older than the cutoff time.

Closes #918.

Example promtail config for getting up to the last 12h of journal entries:

```yaml
server:
  http_listen_port: 9080
  grpc_listen_port: 0

positions:
  filename: /tmp/positions.yaml

clients:
  - url: http://localhost:3100/api/prom/push

scrape_configs:
- job_name: journal
  journal:
    max_age: 12h
    path: /var/log/journal
    labels:
      job: systemd-journal
  relabel_configs:
    - source_labels: ['__journal__systemd_unit']
      target_label: 'unit'
```

**Checklist**
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by slim-bean** [comment]
Should we call this field "since" which would be consistent with the query api?

**Comment by slim-bean** [comment]
hrm on second thought I don't like "since", what about "max_age"?

**Comment by rfratto** [comment]
`max_age` works for me and seems a little more intuitive than `cutoff`, will push new commit soon

File: pkg/promtail/targets/journaltarget_test.go [hunk 1]
```diff
@@ -110,3 +116,124 @@ func TestJournalTarget(t *testing.T) {
...
+
+	initRandom()
+	dirName := "/tmp/" + randName()
+	positionsFileName := dirName + "/positions.yml"
+
+	// Set the sync period to a really long value, to guarantee the sync timer
+	// never runs, this way we know everything saved was done through channel
+	// notifications when target.stop() was called.
+	ps, err := positions.New(logger, positions.Config{
+		SyncPeriod:    10 * time.Second,
```
Comment (slim-bean) [inline pkg/promtail/targets/journaltarget_test.go]: lol, is 10 seconds _really long_ by circle ci standards?

Comment (rfratto) [reply pkg/promtail/targets/journaltarget_test.go]: hah, I did copy this from another test, but 10 seconds should still be appropriate here I think, unless we've seen individual unit tests taking >10 seconds

**Pull request merged**